src/number_theory/function_field.lean

```diff
@@ -107,4 +109,106 @@ is_integral_closure.is_dedekind_domain Fq[X] (ratfunc Fq) F _
 
 end ring_of_integers
 
+/-! ### The place at infinity on Fq(t) -/
+
+section infty_valuation
+
+variable [dec : decidable_eq (ratfunc Fq)]
+include dec
+/-- The valuation at infinity is the nonarchimedean valuation on `Fq(t)` with uniformizer `1/t`.
+Explicitly, if `f/g ∈ Fq(t)` is a nonzero quotient of polynomials, its valuation at infinity is
+`multiplicative.of_add(degree(f) - degree(g))`. -/
+def infty_valuation_def (r : ratfunc Fq) : with_zero (multiplicative ℤ) :=
```

Comment:
If you introduce the degree of a rational function, you can state the basic results below in its API (I mean, the corresponding results). It seems to a me a nicer approach (and also more general, these have noting to do with `Fq`, right?).

Reply:
In the `function_field.lean` file, `Fq` is any field (the name is misleading, there's no finiteness hypothesis on it, but maybe this could be changed in a different PR). Are you suggesting to allow the coefficients to be any commutative ring?

Reply:
Not really, I thought `Fq` were a finite field. But if you write an API for the degree you can try to have `comm_ring` if that works without problem, but I wouldn't spend too much time on it.

Reply:
I rewrote `infty_valuation` in terms of `int_degree`. I only defined `int_degree` for `ratfunc K`, with `K` a field (since `num` and `denom` where only defined in that context).